### ISSUE TITLE
`additive-expr` doesn't work with some user defined subtypes of `int`, `float`, `decimal`

### ISSUE THREAD

**Issue by lasinicl** [ballerina-platform/ballerina-lang#32828]
**Description:**
$title

**Steps to reproduce:**
1.
```ballerina
type IntType1 -2|-1|0|1|2;

type IntType2 int:Unsigned8|int:Signed32;

type IntType3 IntType1|IntType2;

public function main() {
    int a = 123;
    IntType3 b = 2;
    int c = a + b; // operator '+' not defined for 'int' and 'IntType3'
}
```
This should work since both the operands are subtypes of `int`

-----------------------------------------------------------------------------------------------------------

2.
```ballerina
const float A = 1.25;
const float B = 2.0;

type FloatType1 A|B;

public function main() {
    A a = 1.25;
    float|FloatType1 b = 25346.45f;

    float x = a + b; // operator not defined
}
```
This should work since both the operands are subtypes of `float`

----------------------------------------------------------------------------------------------

3.
```ballerina
const decimal A = 1.25;
const decimal B = 2.0;

type DecimalType1 A|B;


public function main() {
    A a = 1.25;
    decimal|DecimalType1 b = 25346.45d;

    decimal x = a + b; // operator not defined
}
```
This should work since both the operands are subtypes of `decimal`

**Affected Versions:**
slbeta2

**Comment by KRVPerera** [comment]
#### TODO
- [ ] https://github.com/ballerina-platform/ballerina-lang/issues/33265

**Comment by MaryamZi** [comment]
Fixed in #33614, merged to master in #33977.
